server/sonar-server/src/test/java/org/sonar/server/component/ws/SearchProjectsActionTest.java

```diff
@@ -918,6 +921,28 @@ public void sort_by_quality_gate_then_by_name() throws Exception {
   }
 
   @Test
+  public void sort_by_last_analysis_date() throws Exception {
+    userSession.logIn();
+    OrganizationDto organization = db.organizations().insert();
+    ComponentDto project1 = db.components().insertPublicProject(organization);
+    authorizationIndexerTester.allowOnlyAnyone(project1);
+    ComponentDto project2 = db.components().insertPublicProject(organization);
+    db.components().insertSnapshot(project2, snapshot -> snapshot.setCreatedAt(40_000_000_000L).setLast(true));
+    authorizationIndexerTester.allowOnlyAnyone(project2);
+    ComponentDto project3 = db.components().insertPublicProject(organization);
+    db.components().insertSnapshot(project3, snapshot -> snapshot.setCreatedAt(20_000_000_000L).setLast(true));
+    authorizationIndexerTester.allowOnlyAnyone(project3);
+    ComponentDto project4 = db.components().insertPublicProject(organization);
+    db.components().insertSnapshot(project4, snapshot -> snapshot.setCreatedAt(10_000_000_000L).setLast(false));
+    db.components().insertSnapshot(project4, snapshot -> snapshot.setCreatedAt(30_000_000_000L).setLast(true));
+    authorizationIndexerTester.allowOnlyAnyone(project4);
+    projectMeasuresIndexer.indexOnStartup(null);
+
+    assertThat(call(request.setSort(ANALYSIS_DATE).setAsc(true)).getComponentsList()).extracting(Component::getKey)
```

Comment:
I think we should also test by descending sort (by only duplicating this line with setAsc(false) and change expected project keys)